docs/cudf/source/developer_guide/testing.md

```diff
@@ -0,0 +1,143 @@
+# Testing cuDF
+
+## Tooling
+Tests in cuDF are written using [`pytest`](https://docs.pytest.org/en/latest/).
+Test coverage is measured using [`coverage.py`](https://coverage.readthedocs.io/en/6.4.1/),
+specifically the [`pytest-cov`](https://github.com/pytest-dev/pytest-cov) plugin.
+Code coverage reports are uploaded to [Codecov](https://app.codecov.io/gh/rapidsai/cudf).
+Each PR also indicates whether it increases or decreases in test coverage.
+
+## Test organization
+
+How tests are organized depends on which of the following two groups they fall into:
+
+1. Free functions operating on the above classes like `cudf.merge`.
+2. Methods of classes like `DataFrame` or `Series`.
+
+Tests of free functions should be grouped into files based on the API sections in our documentation.
+This places tests of similar functionality into the same testing module.
+Tests of class methods should be organized in the same way, except that this organization should be within a subdirectory corresponding to the class.
+For instance, tests of `DataFrame` indexing should be placed into `dataframe/test_indexing.py`.
+In cases where tests may be shared by multiple classes sharing a common parent (e.g. `DataFrame` and `Series` both require `IndexedFrame` tests),
+the tests may be placed in a directory corresponding to the parent class.
+
+## Test contents
+
+### Writing tests
+
+In general, functionality must be tested both for standard use cases and exceptional cases.
+Standard use cases may be covered using parametrization (using `pytest.mark.parametrize`).
+Tests of standard use cases should typically include some coverage of:
+- Different dtypes, including nested dtypes (especially strings)
+- Mixed objects, e.g. binary operations between `DataFrame` and `Series`
+- Operations on scalars
+- Verifying all combinations of parameters for complex APIs like `cudf.merge`.
+
+Here are some of the most common exceptional cases to test:
+1. `Series`/`DataFrame`/`Index` with zero rows
+2. `DataFrame` with zero columns
+3. All null data
+4. For string or list APIs, empty strings/lists
+5. For list APIs, lists containing all null elements or empty strings
+6. For numeric data:
+  1. All 0s.
+  2. All 1s.
+  3. Containing/all inf
+  4. Containing/all nan
+  5. `INT${PRECISION}_MAX` for a given precision (e.g. `2**32` for `int32`).
+7. For strings, non-`en_US` locale.
```

Comment:
Do you mean non-ASCII (various UTF-8) characters? If not: Where does the locale play a role? I am not aware of us testing with different locales.

Reply:
This came from @mroeschke's [previous comment](https://github.com/rapidsai/cudf/pull/11199#discussion_r919206414). Matt, could you comment further? This may not be super relevant for cudf's current feature set.

Reply:
Yeah I am not sure what level of support cudf has for non-`en_US` locale, but pandas has CI builds running our test suite while setting the locale to `zh_CN` and `it_IT`. 

This effects functionality like date & time parsing, encoding & string ops in pandas which has some support for non-`en_US` locales.

Reply:
I think that locale settings are irrelevant / ignored by cuDF but I am not certain.

Reply:
I think you are correct and I will remove this for now.